src/State/Scope.php

```diff
@@ -130,17 +130,39 @@ public function setExtras(array $extras): self
     /**
      * Sets the given data in the user context.
      *
-     * @param array $data The data
+     * @param array $data  The data
+     * @param bool  $merge If true, $data will be merged into user context instead of replacing it
      *
      * @return $this
      */
-    public function setUser(array $data): self
+    public function setUser(array $data, bool $merge = false): self
     {
+        if ($merge) {
+            $this->user->merge($data);
+
+            return $this;
+        }
+
+        @trigger_error('Replacing user context on setUser is deprecated since sentry-php 2.3 and will be changed to a merge on 3.0. '
```

Comment:
Please make this function call one-line and replace the text with ``Replacing the data is deprecated since version 2.3 and will stop working from version 3.0. Set the second argument to `true` to merge the data instead.``

Reply:
0f2e231

Reply:
whoops 3482376